flows/src/main/java/com/softwaremill/jox/flows/Flow.java

```diff
@@ -78,20 +84,32 @@ public Flow(FlowStage<T> last) {
     // region Run operations
 
     /** Invokes the given function for each emitted element. Blocks until the flow completes. */
-    public void runForeach(Consumer<T> sink) throws Exception {
-        last.run(sink::accept);
+    public void runForeach(ThrowingConsumer<T> sink) throws Exception {
+        try {
+            last.run(sink::accept);
+        } catch (JoxScopeExecutionException e) {
+            throw (Exception) e.getCause();
```

Comment:
sorry but I still don't understand why this is the correct thing to do

if a `JoxScopeExecutionException` is thrown, it means there was a scope, which ended with an exception. The root cause for this might have been in an asynchronously running thread. Then, the JSEE will contain the fragment of the stack trace for this thread: including this invocation, all the way up to the point where the scope was joined. The cause, on the other hand, might be from an entirely different thread, and won't even contain this invocation. It seems to me, we're loosing an important piece of context?

Reply:
If we don't want to unwrap jox execution exception, we probably need to make it explicit in java docs

Reply:
The problem with unwrapping is that you loose part of the stack trace: you know that a RE exception was thrown in some thread, but you don't know from where it was invoked - and that's often valuable.

I think there are two solutions:
* leave the exceptions wrapped as-is
* unwrap, but add the original as suppressed

Reply:
That's true, adding original exception as suppressed might be even more blurry. 

I'll revert changes of unwrapping and I'll add some notes to the documentation

Reply:
Ok, reverted